end_to_end_tests/test_end_to_end.py

```diff
@@ -168,18 +145,17 @@ def test_literal_enums_end_to_end():
     )
 )
 def test_meta(meta: str, generated_file: Optional[str], expected_file: Optional[str]):
```

Comment:
I've rewritten this and other functions in this file to use the same `generate_client` context manager, even though the logic in the tests is fairly simple, for two reasons:
1. It enforces consistent state cleanup. Previously, failures in some of these tests would leave generated files behind in the project directory.
2. I think it makes the tests a bit easier to read by reducing boilerplate, so what's left is the logic specific to that test.